src/udisksdaemonutil.c

```diff
@@ -1616,3 +1617,74 @@ udisks_ata_identify_get_word (const guchar *identify_data, guint word_number)
  out:
   return ret;
 }
+
+
+/* Linux kernel version check */
+
+static guint64 detected_linux_version;
+
+/**
+ * udisks_find_linux_version:
+ *
+ * Retrieves version of currently running linux kernel and stores it for
+ * subsequent use. Acts as an initializer for the cached value.
+ *
+ * Returns: detected Linux kernel version in the "MAJOR*1000000 + MINOR*1000 + MICRO"
+ *          format.
+ */
+guint64
+udisks_find_linux_version (void)
+{
+  struct utsname buf = {0,};
+  int n;
+  int major = 0;
+  int minor = 0;
+  int micro = 0;
+  int x;
+  guint64 ver = 0;
+
+  g_warn_if_fail (detected_linux_version == 0);
+
+  if (uname (&buf))
+    {
+      udisks_warning ("Failed to get kernel version: %m");
+      return 0;
+    }
+
+  if (g_ascii_strncasecmp (buf.sysname, "Linux", sizeof buf.sysname) != 0)
+    return 0;
+
+  n = sscanf (buf.release, "%d.%d.%d%n", &major, &minor, &micro, &x);
```

Comment:
Why the `%n` part and `&x` that is not checked?

Reply:
Hmm, I'm not sure :-) The intention was to stop parsing for the rest of the string but it seems it works just fine without it. Removed.